cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,893 @@
+    - overlooked an object (missing annotated bounding box),
+    - chose the wrong class label for an annotated box in the correct location,
+    - imperfectly annotated the location/edges of a bounding box.
+    Any of these annotation errors should lead to an image with a lower label quality score.
+
+    Score is between 0 and 1.
+
+        - 1 - clean label (given label is likely correct).
+        - 0 - dirty label (given label is likely incorrect).
+
+    A score is calculated for each of ``N`` images, with ``K`` total classes in the data.
+    Each image has ``L`` annotated bounding boxes and ``M`` predicted bounding boxes.
+
+    Parameters
+    ----------
+    labels:
+        A list of ``N`` dictionaries such that ``labels[i]`` contains the given labels for the `i`-th image.
+        Refer to documentation for this argument in :py:func:`find_label_issues <cleanlab.object_detection.filter.find_label_issues>` for further details.
+
+    predictions:
+        A list of ``N`` ``np.ndarray`` such that ``predictions[i]`` corresponds to the model predictions for the `i`-th image.
+        Refer to documentation for this argument in :py:func:`find_label_issues <cleanlab.object_detection.filter.find_label_issues>` for further details.
+
+    aggregation_weights:
+       Optional dictionary to specify weights for aggregating quality scores for subtype of label issue into an overall label quality score for the image.
+       Its keys are: "overlooked", "swap", "badloc", and values should be nonnegative weights that sum to 1.
+       Increase one of these weights to prioritize images with bounding boxes that were either:
+       missing in the annotations (overlooked object), annotated with the wrong class label (class for the object should be swapped to another class), or annotated in a suboptimal location (badly located).
+
+       swapped examples, bad location examples, and overlooked examples.
+       It is important to ensure that the weights are non-negative values and that their sum equals 1.0.
+    verbose : bool, default = True
+      Set to ``False`` to suppress all print statements.
+
+    Returns
+    ---------
+    label_quality_scores:
+        Array of shape ``(N, )`` of scores between 0 and 1, one per image in the object detection dataset.
+        Lower scores indicate images that are more likely mislabeled.
+    """
+    method = "objectlab"
+    probability_threshold = 0.0
+
+    _assert_valid_inputs(
+        labels=labels,
+        predictions=predictions,
+        method=method,
+        threshold=probability_threshold,
+    )
+    aggregation_weights = _get_aggregation_weights(aggregation_weights)
+
+    return _compute_label_quality_scores(
+        labels=labels,
+        predictions=predictions,
+        method=method,
+        threshold=probability_threshold,
+        aggregation_weights=aggregation_weights,
+        verbose=verbose,
+    )
+
+
+def _assert_valid_aggregation_weights(aggregation_weights):
```

Comment:
Should this (+ any other assertion functions) be moved to `internal/object_detection_utils.py` ?